Jasmine/Common/Models/TextGrid.swift

```diff
@@ -0,0 +1,61 @@
+struct TextGrid {
+    /// The actual grid.
+    private var grid: [[String?]]
+
+    /// Initializes a CharacterGrid, given the initial grid.
+    ///
+    /// - Parameters:
+    ///   - initialGrid: the initial grid. The CharacterGrid will read from this grid.
+    ///   - randomized: pass in true if the resulting grid wants to be randomized from the initial grid.
+    init(fromInitialGrid initialGrid: [[String]], randomized: Bool) {
+        let numRows = initialGrid.count
+        let numColumns = initialGrid.first?.count ?? 0
+        assert(numRows > 0 && numColumns > 0, "Number of numRows and columns should be more than 0")
+        assert(initialGrid.map { $0.count }.isAllSame, "All numRows should have the same length")
+
+        if randomized {
+            let shuffledGrid = initialGrid.joined().shuffled()
+
+            grid = (0..<numRows).map { row in
+                (0..<numColumns).map { col in
+                    shuffledGrid[row * numColumns + col]
+                }
+            }
+        } else {
+            grid = initialGrid
+        }
+    }
+
+    /// Gets the element in the specified coordinate.
+    ///
+    /// - Parameter coordinate: the coordinate of the element
+    subscript(coordinate: Coordinate) -> String? {
```

Comment:
Apparently you still access it with `blah[coordinate]`. You never put labels in subscripts.